tests/Unit/Parallel/PhaseControl/Test_ExecutePhaseChange.cpp

```diff
@@ -0,0 +1,304 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include "Parallel/PhaseControl/ExecutePhaseChange.hpp"
+#include "Parallel/PhaseControl/PhaseChange.hpp"
+#include "Parallel/PhaseControl/PhaseControlTags.hpp"
+#include "ParallelAlgorithms/EventsAndTriggers/Trigger.hpp"
+#include "Utilities/Functional.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace {
+
+namespace TestTags {
+template <PhaseControl::ArbitrationStrategy, size_t Index>
+struct Request {
+  using type = bool;
+
+  using combine_method = funcl::Or<>;
+  using main_combine_method = funcl::Or<>;
+};
+}  // namespace TestTags
+
+template <PhaseControl::ArbitrationStrategy Strategy, size_t Index,
+          typename PhaseChangeRegistrars>
+struct TestPhaseChange;
+
+namespace Registrars {
+template <PhaseControl::ArbitrationStrategy Strategy, size_t Index>
+struct TestPhaseChange {
+  template <typename PhaseChangeRegistrars>
+  using f = ::TestPhaseChange<Strategy, Index, PhaseChangeRegistrars>;
+};
+}  // namespace Registrars
+
+template <PhaseControl::ArbitrationStrategy Strategy, size_t Index,
+          typename PhaseChangeRegistrars =
+              tmpl::list<Registrars::TestPhaseChange<Strategy, Index>>>
+struct TestPhaseChange : public PhaseChange<PhaseChangeRegistrars> {
+  /// \cond
+  TestPhaseChange() = default;
+  explicit TestPhaseChange(CkMigrateMessage* /*unused*/) noexcept {}
+  using PUP::able::register_constructor;
+  WRAPPED_PUPable_decl_template(TestPhaseChange);  // NOLINT
+  /// \endcond
+  static std::string name() noexcept {
+    if constexpr (Strategy ==
+                  PhaseControl::ArbitrationStrategy::RunPhaseImmediately) {
+      if constexpr(Index == 0_st) {
```

Comment:
The formatting looks off here and in some instances below.. maybe clang-format this file?